File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.empty()){
return " ";}
        sort(strs.begin(),strs.end());
        string f=strs[0];
        string l=strs[strs.size()-1];
        string p="";
        
    for (int i = 0; i < min(f.size(), l.size()); i++) {
        if (f[i] == l[i]) {
            p += f[i];
        } else {
            break;
        }
    
    }
    return p;


    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
public class Solution {
    public static String longestCommonPrefix(String[] strs) {

        if (strs == null || strs.length == 0)
            return "";

        String prefix = "";
        String first = strs[0];

        for (int i = 0; i < first.length(); i++) {
            char ch = first.charAt(i);

            for (int j = 1; j < strs.length; j++) {
                if (i >= strs[j].length() || strs[j].charAt(i) != ch) {
                    return prefix;
                }
            }

            prefix = prefix + ch;
        }

        return prefix;
    }
}
